set-web/src/main/java/eu/europeana/set/web/service/controller/BaseRest.java

```diff
@@ -188,6 +184,29 @@ public LdProfiles getProfile(String paramProfile, HttpServletRequest request) th
 		return profile;
 	}
 
+	/**
+	 * This method identifies profile from a HTTP header if it exists.
+	 * 
+	 * @param paramProfile
+	 *            The HTTP request parameter
+	 * @param request
+	 *            The HTTP request with headers
+	 * @return profile value
+	 * @throws HttpException
+	 * @throws UserSetProfileValidationException
+	 */
+	public LdProfiles getHeaderProfile(HttpServletRequest request) throws HttpException {
+
+		LdProfiles profile = null;
+		String preferHeader = request.getHeader(HttpHeaders.PREFER);
+		if (preferHeader != null) {
+			// identify profile by prefer header
+			profile = getProfile(preferHeader);
```

Comment:
we don't need two different methos getHeaderProfile and getProfile

Reply:
combined in one method